hack/update-deps.sh

```diff
@@ -48,9 +47,21 @@ readonly GO_GET
 
 if (( GO_GET )); then
   go get -d ${FLOATING_DEPS[@]}
+
+  # Update test-infra
+  # TODO(coryrc): move this to a file in knative/test-infra/scripts
+  git remote add test-infra https://github.com/knative/test-infra.git || true
+  git fetch test-infra master
+  # First-time setup:
+  #  mkdir -p scripts/test-infra
+  #  git read-tree --prefix=scripts/test-infra -u test-infra/master:scripts
+  # Each update:
+  git rm -fr scripts/test-infra
+  git read-tree --prefix=scripts/test-infra -u test-infra/master:scripts
```

Comment:
Is it always that the latest master is stable ? When doing a release we are pinning the eventing/serving version to a version number. Is test-infra also doing release versions ? or is it just a continuous integration flow ? (side question: How can historical builds be reproduced the ?) 

Can we add at least a commit hash to the checked-out scripts so that we have a reference, something like single-lined `scripts/test-infra/COMMIT`

Reply:
This was a bit of a hack, I made a real version here:
https://github.com/knative/test-infra/pull/2068

Reply:
I'm blocked from deploying though: https://github.com/knative/test-infra/issues/2069

Reply:
How would one bootstrap the updarte file in knative/test-infra#2068 ? I wouldn't mind to add this script already to the client archive (our release is one week after serving/eventing, so enough time to test) now, but I would also like to add the call to this update script into build.sh (the single entrypoint for developers).

Reply:
It assumes it is 2 levels deep, the intention was to keep in knative/test-infra/scripts directory and run from scripts/test-infra so it self-updates, but given it isn't there yet, you could:

Copy it to some directory two levels deep (uhh, `hack/update/update-test-infra.sh`?). `git add` it. Then run `./hack/update/update-test-infra.sh --first-time` from the repo root and you've in.

Reply:
I did that for this PR

Reply:
I got the PR approved, so I updated this PR to use the script the appropriate way. Feel free to copy this method for client-contrib or here as you feel fit.